apps/dialtone-documentation/docs/.vuepress/baseComponents/CodeExampleTabs.vue

```diff
@@ -57,26 +57,67 @@
 </template>
 
 <script setup>
+import { computed, onMounted, ref, nextTick } from 'vue';
 import Prism from 'prismjs';
+import pretty from 'pretty';
 import CopyButton from './CopyButton.vue';
-import { ref } from 'vue';
 import { getUniqueString } from '@workspaceRoot/common/utils';
 
 const props = defineProps({
+  /**
+   * Retrieves the reference to the example component to get the HTML code.
+   * If not provided, the HTML code will be used.
+   *
+   * @returns {Component} The reference to the component.
+   */
+  getComponentRef: {
+    type: Function,
+    default: null,
+  },
```

Comment:
I like this addition.
But now we have two ways to do sth internal.

On our `code-example-tabs`, we could use
`getComponentRef` that you are using passing `ref=` on the `code-well-header` section.
or
`htmlCode` 

As I see there are some pro and cons on this.

cons
- since this is internal development, I dont see why we should have two different ways to achieve this.
kind of confusing when you should use one or another, I will probably use on those components that generates large html code but in these cases we just add a `...` and thats it avoid repited code, like datepicker generates a bunch of similar code.
- we dont have the ability to modified sth on the generated html code, now we are using regex. This can be less flexible in some particular case.
- Seems like it will work only on single components, so if our example has multiple we cannot use it.
Maybe we can improve it with a wrapper, dunno.
- It is necessary the addition of the pretty dependency
- In some cases seems like it is not doing the right job 

<img width="927" alt="image" src="https://github.com/dialpad/dialtone/assets/89984179/6bd28421-9312-4126-9931-3d135c6998b6">


pros
- Much less code and less style error prone (indents, close html tags,etc..)
- Probably we can use it if we improve it, in most of the cases, if this is the case I would agree on have it both ways, for example if we could use it in the 90% of the cases. So `htmCode` will be used in particular cases.


What I would suggest, move this to a different PR, with a different ticket as a proposal to improvement and we can continue this discussion and implementation.

Reply:
If I'm missing sth on the idea of implement it, let me know, maybe Im skipping some advantage of it.

Reply:
The new prop `getComponentRef` allows for an "autogenerated" html tab. It has a lot of advantages like it's honest, and it will make it easier to migrate everything from the following tickets in this epic.
The advantage of having both ways of generating the html code is that we don't need to migrate everything we already have there.
It works if there are many components in the example if you add the `ref` to the wrapper.
I don't see any problems really in adding this in this PR. It is a third party dependency needed for formatting, but it's only a dev dependency and I haven't found any problems with it.

Reply:
> In some cases seems like it is not doing the right job

Would prettier work better? It's considered the standard for opinionated code formatting. Might have a lot more options we can work with.
https://www.npmjs.com/package/prettier

Reply:
I changed it to use prettier, although it has the same caveats as pretty, but it is more standardised and also might be useful for code formatting in the future.